MetaMask/metamask-mobile
# fix: accounts not syncing between devices bug

## **Description**

This PR fixed an issue preventing mobile accounts to sync its states (enable/disable) with other clients (platform and extension). [NOTIFY-1218](https://consensyssoftware.atlassian.net/browse/NOTIFY-1218?atlOrigin=eyJpIjoiZTRlYTlkNjVjOGVjNDAyYTljNmM3YTg2NjBiYzllNGYiLCJwIjoiaiJ9)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Import the same SRP on Mobile
2. Make sure all accounts are visible on both devices/platforms. If not, import all accounts
3. Switch Notifications state in account "A" on extension.
4. Go to Notifications Settings on mobile and see if the state is propagated.
5. Switch back on mobile
6. Go to extension and see if state is propagated. 

## **Screenshots/Recordings**



### **Before**



### **After**

https://github.com/user-attachments/assets/860e2e46-f08a-45fd-b965-a9948ef59630




## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**



[NOTIFY-1218]: https://consensyssoftware.atlassian.net/browse/NOTIFY-1218?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
